src/SB/Core/x/xParEmitter.cpp

```diff
@@ -1,3 +1,828 @@
+    {
+        temp_oocull = 1.0f / pea->cull_dist_sqr;
+    }
+    else
+    {
+        temp_oocull = 0.0f;
+    }
+    bEmitter->oocull_distance_sqr = temp_oocull;
+    bEmitter->emit_volume = 0;
+    if ((pea->emit_type == 0xE) || (pea->emit_type == 0xD))
+    {
+        vec_length = (pea->e_circle.dir).length2();
+        if (vec_length < 0.001f)
+        {
+            pea->e_circle.dir = 0;
+        }
+        else
+        {
+            pea->e_circle.dir = 1.0f / xsqrt(vec_length);
+        }
+    }
+    bEmitter->last_attach_loc += pea->e_circle.dir;
+}
+
+void xParEmitterSetup(xParEmitter* t)
+{
+    xParEmitterAsset* pea;
+    xBase* temp_base;
+
+    if (t->parSys != (xParSys*)0x0)
+    {
+        t->group = *(xParGroup**)(t->parSys + 0x20);
+    }
+    if ((t->tasset->attachToID != 0) && (t->tasset->emit_type == '\t'))
+    {
+        temp_base = zSceneFindObject(t->tasset->attachToID);
+        iModelTagSetup(&t->tag, *(RpAtomic**)(*(S32*)&temp_base[2].baseType + 0x10),
+                       *(F32*)&t->tasset->e_entbound, (t->tasset->e_entbound).expand,
+                       (t->tasset->e_entbound).deflection);
+    }
+    temp_base = (xBase*)0x0;
+    if (t->tasset->attachToID != 0)
+    {
+        temp_base = zSceneFindObject(t->tasset->attachToID);
+    }
+    t->attachTo = temp_base;
+    if (((t->tasset->emit_type != '\x10') && (t->tasset->emit_type != '\x0F')) &&
+        (t->tasset->emit_type == '\x06'))
+    {
+        temp_base = zSceneFindObject(t->tasset->attachToID);
+        t->emit_volume = temp_base;
+    }
+    add_tweaks(*t);
+    return;
+}
+
+void xParEmitterReset(xParEmitter* t)
+{
+    xBaseReset((xBase*)t, t->tasset);
+    t->emit_flags = t->tasset->emit_flags;
+}
+
+S32 xParEmitterEventCB(xBase* to1, xBase* to2, U32 toEvent, F32* unused1, xBase* unused2)
+{
+    xParEmitterCustomSettings sp8;
+
+    switch ((S32)toEvent)
+    {
+    case 0xA:
+        xParEmitterReset((xParEmitter*)to2);
+        break;
+    case 0x26:
+        *(U8*)(toEvent + 0x30) |= 1;
+        break;
+    case 0x27:
+        if (*(U8*)(toEvent + 0x30) & 1)
+        {
+            *(U8*)(toEvent + 0x30) = *(U8*)(toEvent + 0x30) ^ 1;
+        }
+        break;
+    case 0xFA:
+        memset(&sp8, 0, 0x16C);
+        xParEmitterEmitCustom((xParEmitter*)to2, 0.033333335f, &sp8);
+        break;
+    }
+    return 1;
+}
+
+xPar* xParEmitterEmitCustom(xParEmitter* p, F32 dt, xParEmitterCustomSettings* info)
+
+{
+    U8 emit_type;
+    U32 custom_flags;
+    xParEmitterAsset* p_tasset;
+
+    custom_flags = info->custom_flags;
+    p_tasset = p->tasset;
+    if ((custom_flags & 1) != 0)
+    {
+        sSaveEmmiterSettings(p_tasset, p_tasset, 0.0);
+        sSaveEmmiterPropSettings(p->prop, p->prop, 0.0);
+    }
+    if ((custom_flags & 0x100) != 0)
+    {
+        p_tasset->pos = info->pos;
+    }
+    if ((custom_flags & 0x10) != 0)
+    {
+        p->prop->value[0] = info->value[0];
+    }
+    if ((custom_flags & 2) != 0)
```

Comment:
Please make the integer flag comparisons consistent with the other hex values by making them hex values to represent the fact that bits are being compared.